cleanlab/filter.py

```diff
@@ -735,7 +510,7 @@ class 0, 1, ..., K-1.
       labeled with high confidence.
     """
 
-    assert len(pred_probs) == len(labels)
+    assert_valid_inputs(X=None, y=labels, pred_probs=pred_probs, multi_label=multi_label)
```

Comment:
how much slower is the package by checking the extra things checked by `assert_valid_inputs` -- can you run a few things before and after and %timeit?

Reply:
i would assume so, too. It just makes reviewing PRs like this a lot easier if you're able to share the time to run pytest tests/ before and after this PR

Side note -- we should definitely add that feature to the CI & CD sometime soon. @ryansingman  @anishathalye

Reply:
Btw, talking about just asymptotics doesn't really capture concrete running time that well. We should be using concrete numbers from benchmarks.

Also, the `(pred_probs < 0)` allocates a M x N array, and the `(pred_probs > 1)` allocates a whole other M x N array. This can be done in O(1) space instead.

Reply:
Sorry I wasn't referring to asymptotics when i mentioned: in practical terms these lines are negligible extra time for the functions in which they are called.

(not very reliable) timing of `pytest` (all unit tests)

user+sys time for existing master branch:  1855.51s
user+sys time for this PR code version:  1844.67s

This is not a good benchmark obviously... firstly because the inputs are small. 
For big inputs it'll obviously take longer since there are two new linear-time checks, but as I mentioned the practical effect is negligible compared to the runtime of the original function in which the checks have been added.